src/panels/lovelace/cards/hui-thermostat-card.ts

```diff
@@ -0,0 +1,309 @@
+import { html, LitElement } from "@polymer/lit-element";
+import { classMap } from "lit-html/directives/classMap.js";
+import roundSliderStyle from "round-slider/dist/roundslider.min.css";
+
+import "../../../components/ha-card.js";
+import "../../../components/ha-icon.js";
+import "../../../resources/jquery.roundslider";
+
+import { HomeAssistant } from "../../../types.js";
+import { HassLocalizeLitMixin } from "../../../mixins/lit-localize-mixin";
+import { LovelaceCard, LovelaceConfig } from "../types.js";
+
+const thermostatConfig = {
+  radius: 150,
+  min: 50,
+  max: 90,
+  step: 1,
+  circleShape: "pie",
+  sliderType: "min-range",
+  value: 70,
+  startAngle: 315,
+  width: 5,
+  lineCap: "round",
+  handleSize: "+10",
+  showTooltip: false,
+};
+
+const modeIcons = {
+  auto: "mdi:autorenew",
+  heat: "hass:fire",
+  cool: "hass:snowflake",
+  off: "mdi:fan-off",
+};
+
+interface Config extends LovelaceConfig {
+  entity: string;
+}
+
+let loaded: Promise<void>;
+
+export class HuiThermostatCard extends HassLocalizeLitMixin(LitElement)
+  implements LovelaceCard {
+  public hass?: HomeAssistant;
+  protected config?: Config;
+  protected roundSlider: any;
+
+  static get properties() {
+    return {
+      hass: {},
+      config: {},
+    };
+  }
+
+  public getCardSize() {
+    return 4;
+  }
+
+  public setConfig(config: Config) {
+    if (!config.entity || config.entity.split(".")[0] !== "climate") {
+      throw new Error("Specify an entity from within the climate domain.");
+    }
+
+    this.config = config;
+  }
+
+  protected render() {
+    const stateObj = this.hass!.states[this.config!.entity];
+    const sizeClass = this.clientWidth > 450 ? "large" : "small"; // Use Class Map
+    return html`
+      ${this.renderStyle()}
+      <ha-card
+        class="${stateObj.attributes.operation_mode} ${sizeClass}">
+        <div id="root">
+          <div
+            .hass="${this.hass}"
+            .config="${this.config}"
+            .localize="${this.localize}"
```

Comment:
Why?

Reply:
I don't get any of these properties here.

Reply:
I dont either... 🤔 I was tired... (does that work as an excuse for how bad that is 😆 )